### PR TITLE
add JWT authentication support & login endpoint

### PR DESCRIPTION
**Pull request by VoitseshukV**

### CODE REVIEW

File: src/main/java/com/bookstore/core/dto/CreateBookRequestDto.java [hunk 1]
```diff
@@ -5,24 +5,23 @@
...
-    @Positive
-    private BigDecimal price;
-    private String description;
-    private String coverImage;
+public record CreateBookRequestDto(
+        @NotBlank
+        String title,
+        @NotBlank
+        String author,
+        @NotNull
```
Comment (kozhukhovsky) [inline src/main/java/com/bookstore/core/dto/CreateBookRequestDto.java:14]:
```suggestion
public record CreateBookRequestDto(
    @NotBlank
      String title,
      @NotBlank
      String author,
      @NotNull
```

4 spaces need

**Review by kozhukhovsky — APPROVED** [review]
fix codestyle before merge

File: src/main/java/com/bookstore/core/dto/CreateBookRequestDto.java (hunk 1 above)
Comment (VoitseshukV) [reply src/main/java/com/bookstore/core/dto/CreateBookRequestDto.java:14]: @kozhukhovsky It's record fields. Don't they need 8 spaces? They are in block (...), it seems it counts "Continuation indent".

Comment (VoitseshukV) [reply src/main/java/com/bookstore/core/dto/CreateBookRequestDto.java:14]: 2023-11-23 13_00_27-bookstore – CreateBookRequestDto java
Yes, the checkstyle plugin does not like this either

**Pull request merged**